pkg/admin/utils/lazy.go

```diff
@@ -0,0 +1,40 @@
+package utils
+
+type Lazy struct {
+	init  func() (interface{}, error)
+	value interface{}
+	err   error
+}
+
+func NewLazy(init func() (interface{}, error)) *Lazy {
+	return &Lazy{init: init}
+}
+
+func NewLazyValue(value interface{}) *Lazy {
+	switch value := value.(type) {
+	case *Lazy:
+		return value
+	case func() (interface{}, error):
+		return NewLazy(value)
+	default:
+		return &Lazy{value: value}
```

Comment:
Add a case to support making a lazy value of error?

Reply:
This function is like `Promise.resolve`, so it should not try to fill in the error field.

I've added a `NewLazyError` function before, but there's no use so I removed it.